diagnostics/model_test/verify_model.py

```diff
@@ -133,44 +182,68 @@ def parse_json():
+        random_input = np.random.randint(
+            rand_val_range, size=[bs] + dim).astype('float32')
         input_tensor_dim_map[name] = random_input
 
-    # Run the model on tensorflow
+    # Run the model on device1
     result_tf_graph_arrs, out_tensor_names_cpu = calculate_output(
-        parameters, "CPU", input_tensor_dim_map)
-    # Run the model on ngraph
+        parameters, device1, input_tensor_dim_map)
+    # Run the model on device2
     result_ngraph_arrs, out_tensor_names_ngraph = calculate_output(
-        parameters, "NGRAPH", input_tensor_dim_map)
+        parameters, device2, input_tensor_dim_map)
 
     assert all(
         [i == j for i, j in zip(out_tensor_names_cpu, out_tensor_names_ngraph)])
-    l1_norm_threshold = parameters["l1_norm_threshold"]
-    l2_norm_threshold = parameters["l2_norm_threshold"]
-    inf_norm_threshold = parameters["inf_norm_threshold"]
     for tname, result_ngraph, result_tf_graph in zip(
             out_tensor_names_cpu, result_ngraph_arrs, result_tf_graph_arrs):
+        file.write(">>>Start {}\n".format(tname))
+
+        new_out_layer = tname.replace("/", "_")
+        nparray_tf = np.array(result_tf_graph)
+        nparray_ngraph = np.array(result_ngraph)
+        np.save(device1 + "-" + new_out_layer + ".npy", nparray_tf)
+        np.save(device2 + "-" + new_out_layer + ".npy", nparray_ngraph)
+
         l1_norm = calculate_norm(result_ngraph, result_tf_graph, 1)
         l2_norm = calculate_norm(result_ngraph, result_tf_graph, 2)
         inf_norm = calculate_norm(result_ngraph, result_tf_graph, np.inf)
 
         if l1_norm > l1_norm_threshold:
             print("The L1 norm %f is greater than the threshold %f for %s" %
                   (l1_norm, l1_norm_threshold, tname))
+            file.write(
+                "L1 norm test - Fail: The L1 norm %f is greater than the threshold %f\n"
+                % (l1_norm, l1_norm_threshold))
         else:
             print("L1 norm test passed for ", tname)
+            file.write("L1 norm test - Pass\n")
 
         if l2_norm > l2_norm_threshold:
             print("The L2 norm %f is greater than the threshold %f for %s" %
                   (l2_norm, l2_norm_threshold, tname))
+            file.write(
+                "L2 norm test - Fail: The L2 norm %f is greater than the threshold %f\n"
+                % (l2_norm, l2_norm_threshold))
         else:
             print("L2 norm test passed for ", tname)
+            file.write("L2 norm test - Pass\n")
 
         if inf_norm > inf_norm_threshold:
             print("The inf norm %f is greater than the threshold %f for %s" %
                   (inf_norm, inf_norm_threshold, tname))
+            file.write(
+                "Inf norm test - Fail: The inf norm %f is greater than the threshold %f\n"
+                % (inf_norm, inf_norm_threshold))
         else:
             print("inf norm test passed for ", tname)
+            file.write("Inf norm test - Pass\n")
+
```

Comment:
about file.write: It seems it is only writing whatever is being printed to a file. Unless one wants to dump a lot more extra information, we do not need the file.write, as we can just pipe the output to a log file.

my suggestion is that if u need some small info to be printed, u can add a new print line. if u want large chunks to be output, only then we can use the log file. For numbers dump we are doing the .npy output in any case, so perhaps the log file is redundant

Reply:
I agree that the output file is redundant. The idea was if we run many layers then the log screen will scroll up but if the log is saved to a file then it's easy to check whenever we need later. I think it's still ok to have but if this is against code design policy, I'll remove them.

Reply:
We can always pipe the output to a file
```
python verify_model.py --json_file minst.json > log.txt
```